src/jsrs_parser.cc

```diff
@@ -360,6 +360,8 @@ Local<Value> ParseString(Isolate*    isolate,
       char* symb =
           GetControlChar(isolate, begin + ++i, &out_offset, &in_offset);
       if (!symb) {
+        delete[] result;
```

Comment:
There is no point in calling `delete[]` on `nullptr`, and that is what you are doing here.
Also, about TODO: no need to throw an exception, because `GetControlChar` already takes care of this and throws when it returns `nullptr`.